protobuf/protoeditor-core/src/com/intellij/protobuf/lang/psi/PbSyntaxStatementBase.java

```diff
@@ -21,4 +21,7 @@ interface PbSyntaxStatementBase extends PbElement {
 
   @Nullable
   SyntaxLevel getSyntaxLevel();
+
+  @Nullable
+  String getEdition();
```

Comment:
The only thing I find a bit confusing here is that there is a `getEdition` method that might return nothing in case it is an old style file with syntax statement. What if we extend the SyntaxLevel enum so that it would be a sealed class with several implementations. Traditional ones would be Proto2 and Proto3, and the new would be Edition class with custom version text in it?

The idea is identical to what you've proposed, just a bit different implementation

Reply:
Yea that sgtm